packages/idl/index.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('fs').promises;
+const path = require('path');
+
+const WebIDL2 = require('webidl2');
+
+class IDLFile {
+  constructor(dir, file) {
+    this.filename = file;
+    this.shortname = path.basename(file, '.idl');
+    this.path = path.join(dir, file);
+  }
+
+  async text() {
+    const text = await fs.readFile(this.path, 'utf8');
+    return text;
+  }
+
+  async parse() {
+    const text = await this.text();
+    return WebIDL2.parse(text);
```

Comment:
I feel this is somewhat overkill without real uses inside webref, could we just provide the text and let the users do the parsing part in whatever way they want?

Reply:
Note that webidl2 is a peer dependency, and that ensuring that a specific version of webidl2 is able to parse all of the IDL in this package is the main value of a separate package. Are you skeptical of some of that as well, or just saying that the `parse` and `parseAll` helpers here aren't that useful?

Reply:
Ah, my confusion, totally okay if it's a pear dependency.

Reply:
Oh, alright then :)